### PR TITLE
PCHR-2404: Remove default date filter set on the documents dashboard

### PR DESCRIPTION
**Pull request by bmnepali**
## Overview
This PR removes the default date filters the form T&A dashboard Documents section. With some code refactoring to implement new angularjs style guide.

## Before
- Default date ranges are  available and documents are listed after they are filtered.
2404-before-min

## After
- Removed default date filters and the all documents are listed.
2404-after-min

## Technical Details
1. Default  defined dates are set to null, to remove default dates. In file  in `controllers/document-list.js` as follows:
```js
vm.filterParamsHolder = {
     documentStatus: defaultDocumentStatus,
     dateRange: {
         from: null,
         until: null
     }
 };
```
2. Implemented new angularjs style guide for `controllers/document-list.controller.js` with controlerAs syntax in respective template files.
3. Implemented new angularjs style guide for `controllers/task-list.controller.js` with controlerAs syntax in respective template files.
4. Updates respective test cases to use contollerAs syntax.

## Comments
To ensure the respective template files are updated accurately to adope  angular controllerAs syntax following checks have been performed.
1. I have used `grep -R 'List.' views/` to check if  "List." is left unnoticed and not result found!
2. I have also checked using `grep -R "controllerAs: 'List'," js/src/` if any `ControllerAs: List` left in modules and not was found.
3. I have manually checked and verified that in following templates the `list._property_` syntax is applied as required correctly.

> views/dashboard/calendar.documentList.html, views/dashboard/calendar.taskList.html, views/dashboard/documents.html, views/dashboard/documents.html, views/contact/documents.html, and views/dashboard/documents.html

As these (^^) are only the files that get affected when implementing controllerAs syntax in document list controller and task list controller.

4.  I ran following `grep` commands to find if there are any places that are missing the `controllerAs` syntax i.e using `list._property_`. And the following command does not display any suspicious lines or code.
1. `grep -r 'ng-model="[^l][^i][^s][^t]' “path_to_file.html”`
2. `grep -r 'ng-click="[^l][^i][^s][^t]' “path_to_file.html”`
3. `grep -r 'ng-repeat="[^l][^i][^s][^t]' “path_to_file.html”`

- [x] Tests Pass

### CODE REVIEW

File: uk.co.compucorp.civicrm.tasksassignments/views/dashboard/calendar.documentList.html [hunk 1]
```diff
@@ -14,22 +14,22 @@ <h4 class="header-bar header-bar-danger-light">Documents:</h4>
...
     <tbody>
       <tr
-        ng-repeat="document in listPaginated = (listOngoing = (list
+        ng-repeat="document in listPaginated = (listOngoing = (documentList.list
         | filterByStatus:cache.documentStatusResolve:false
         | filterByDue:'dateRange':{ from: calendarDay, until: calendarDay }
         | orderBy: 'activity_date_time')
-        | offset: (pagination.currentPage-1)*pagination.itemsPerPage
-        | limitTo: pagination.itemsPerPage)"
+        | offset: (documentList.pagination.currentPage-1)*documentList.pagination.itemsPerPage
```
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.tasksassignments/views/dashboard/calendar.documentList.html]: Please add spaces around `*`

File: uk.co.compucorp.civicrm.tasksassignments/views/dashboard/calendar.documentList.html [hunk 2]
```diff
@@ -14,22 +14,22 @@ <h4 class="header-bar header-bar-danger-light">Documents:</h4>
...
         src="'document.html'">
       </tr>
     </tbody>
   </table>
   <div class="clearfix"></div>
-  <nav class="row" ng-show="(listOngoing.length > pagination.itemsPerPage)">
+  <nav class="row" ng-show="(listOngoing.length > documentList.pagination.itemsPerPage)">
     <div class="col-xs-12 text-center">
-      <uib-pagination boundary-links="true" total-items="listOngoing.length" rotate="false" items-per-page="pagination.itemsPerPage" max-size="pagination.maxSize" ng-model="pagination.currentPage" class="pagination-sm" previous-text="&lsaquo;" next-text="&rsaquo;" first-text="&laquo;" last-text="&raquo;"></uib-pagination>
+      <uib-pagination boundary-links="true" total-items="listOngoing.length" rotate="false" items-per-page="documentList.pagination.itemsPerPage" max-size="documentList.pagination.maxSize" ng-model="documentList.pagination.currentPage" class="pagination-sm" previous-text="&lsaquo;" next-text="&rsaquo;" first-text="&laquo;" last-text="&raquo;"></uib-pagination>
```
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.tasksassignments/views/dashboard/calendar.documentList.html]: This is a very long line, please break it into a multiline.

File: uk.co.compucorp.civicrm.tasksassignments/views/dashboard/documents.html [hunk 3]
```diff
@@ -30,55 +30,55 @@ <h3 class="panel-title">View Documents By:</h3>
...
               </span>
               </a>
             </div>
           </div>
           <div class="col-xs-12 col-sm-3 text-right">
-            <a href ng-click="isCollapsed.filterAdvanced = !isCollapsed.filterAdvanced" class="btn btn-collapse"><span
+            <a href ng-click="documentList.isCollapsed.filterAdvanced = !documentList.isCollapsed.filterAdvanced" class="btn btn-collapse"><span
               class="fa"
-              ng-class="{fa: true, 'fa-chevron-down': !isCollapsed.filterAdvanced, 'fa-chevron-right': isCollapsed.filterAdvanced }"
+              ng-class="{fa: true, 'fa-chevron-down': !documentList.isCollapsed.filterAdvanced, 'fa-chevron-right': documentList.isCollapsed.filterAdvanced }"
```
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.tasksassignments/views/dashboard/documents.html]: `{fa:` -> `{ fa:`

File: uk.co.compucorp.civicrm.tasksassignments/views/directives/sidebar-filters.html [hunk 4]
```diff
@@ -1,38 +1,38 @@
...
             ).length}}
         </span>
       </a>
     </li>
   </ul>
   <hr>
-  <a href ng-click="isCollapsed.filterDates = !isCollapsed.filterDates" class="btn btn-block {{prefix}}select-dates">
-    <i class="fa" ng-class="{fa: true, 'fa-chevron-down': !isCollapsed.filterDates, 'fa-chevron-right': isCollapsed.filterDates }"></i>
+  <a href ng-click="documentList.isCollapsed.filterDates = !documentList.isCollapsed.filterDates" class="btn btn-block {{prefix}}select-dates">
+    <i class="fa" ng-class="{fa: true, 'fa-chevron-down': !documentList.isCollapsed.filterDates, 'fa-chevron-right': documentList.isCollapsed.filterDates }"></i>
```
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.tasksassignments/views/directives/sidebar-filters.html]: `{fa` -> `{ fa`

File: uk.co.compucorp.civicrm.tasksassignments/views/dashboard/documents.html [hunk 5]
```diff
@@ -140,25 +140,25 @@ <h3 class="panel-title">{{label[filterParams.due]}}</h3>
...
                   src="'document.html'">
                 </tr>
               </tbody>
             </table>
             <div class="clearfix"></div>
-            <nav class="row" ng-show="(listFiltered.length > pagination.itemsPerPage)">
+            <nav class="row" ng-show="(listFiltered.length > documentList.pagination.itemsPerPage)">
               <div class="col-xs-12 text-center">
-                <uib-pagination boundary-links="true" total-items="listFiltered.length" rotate="false" items-per-page="pagination.itemsPerPage" max-size="pagination.maxSize" ng-model="pagination.currentPage" class="pagination-sm" previous-text="&lsaquo;" next-text="&rsaquo;" first-text="&laquo;" last-text="&raquo;"></uib-pagination>
+                <uib-pagination boundary-links="true" total-items="listFiltered.length" rotate="false" items-per-page="documentList.pagination.itemsPerPage" max-size="documentList.pagination.maxSize" ng-model="documentList.pagination.currentPage" class="pagination-sm" previous-text="&lsaquo;" next-text="&rsaquo;" first-text="&laquo;" last-text="&raquo;"></uib-pagination>
```
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.tasksassignments/views/dashboard/documents.html]: Too long, please break.

File: uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/document-list.spec.js [hunk 6]
```diff
@@ -89,9 +89,6 @@ define([
     });
 
     describe('labelDateRange()', function () {
```
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/document-list.spec.js]: Please add tests that check that the default filters are null.

Comment (bmnepali) [inline uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/document-list.spec.js]: adde!

**Comment by bmnepali** [comment]
Fixed merge conflicts by rebase.

**Comment by igorpavlov** [comment]
Please fix a typo in the PR description: `stye` should be `style`.

**Comment by igorpavlov** [comment]
Could you please include GIFs for both Before and After sections showing the whole process of page load, filter change etc since this is exactly what this PR is about.

**Comment by igorpavlov** [comment]
Could you please let me know which latest commits include manual changes and which not?

**Comment by igorpavlov** [comment]
I think there is something wrong with a commit history, for example, this commit https://github.com/compucorp/civihr-tasks-assignments/pull/245/commits/2a5def32cbdd158b5e5c19217b5400488db4bd88 says "renamed" but the file was actually deleted.

File: uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/modules/app-dashboard.module.js [hunk 7]
```diff
@@ -26,8 +26,9 @@ define([
               })
               .state('tasks', {
                 url: '/tasks',
-                controller: 'TaskListCtrl',
+                controller: 'TaskListController',
                 templateUrl: config.path.TPL + 'dashboard/tasks.html?v=' + (new Date().getTime()),
+                controllerAs: 'List',
```
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/modules/app-dashboard.module.js]: I don't really think we use capital letters in controllerAs, this becomes a variable in templates and capitalised variabled usually mean classes.

**Comment by igorpavlov** [comment]
Please remove static screenshots, they are not really needed since there are GIFs now.

**Comment by igorpavlov** [comment]
I think last 4 commits can be squashed, what do you think>

File: uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/modules/app-dashboard.module.js (hunk 7 above)
Comment (bmnepali) [reply uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/modules/app-dashboard.module.js]: Changed to small caps.

File: uk.co.compucorp.civicrm.tasksassignments/views/contact/documents.html [hunk 8]
```diff
@@ -17,7 +17,7 @@
             <div class="form-group" ng-if="settings.extEnabled.assignments">
               <label class="col-xs-12 col-sm-3 control-label">Type:</label>
               <div class="col-xs-12 col-sm-9">
-                <ui-select prevent-animations multiple ng-model="filterParams.assignmentType">
+                <ui-select prevent-animations multiple ng-model="list.filterParams.assignmentType">
```
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.tasksassignments/views/contact/documents.html]: That file didn't contain `List.` before PR Feedback. Did it work at all?

Comment (bmnepali) [reply uk.co.compucorp.civicrm.tasksassignments/views/contact/documents.html]: No it didn't! I didnot notice the change would affect this file as well. I will fix it now.

Comment (igorpavlov) [reply uk.co.compucorp.civicrm.tasksassignments/views/contact/documents.html]: Could you please describe all steps/methods to ensure that all affected files were amended accordingly?

Comment (bmnepali) [reply uk.co.compucorp.civicrm.tasksassignments/views/contact/documents.html]: **Steps taken:**
1. Search every instance of `List._property_` in `views/dashboard/calendar.documentList.html`, `views/dashboard/calendar.taskList.html`, `views/dashboard/documents.html`, `views/dashboard/documents.html`,  `views/contact/documents.html`,  and `views/dashboard/documents.html`.
2. Replaced with new `controllerAs` object `list._property_`
3. Verified the document created/ edited and deleted
4. Verified the task can be created/ edited and deleted

Comment (igorpavlov) [reply uk.co.compucorp.civicrm.tasksassignments/views/contact/documents.html]: Thank you.
1) Do these steps check if the "list" or "List" was not applied at all in templates?
2) Do these steps check **all** templates?

Comment (bmnepali) [reply uk.co.compucorp.civicrm.tasksassignments/views/contact/documents.html]: 1. I have used `grep -R 'List.' views/` to check if  "List." is left unnoticed and not result found!
2. I have also checked using `grep -R "controllerAs: 'List'," js/src/` if any `ControllerAs: List` left in modules and not was found.

Comment (igorpavlov) [reply uk.co.compucorp.civicrm.tasksassignments/views/contact/documents.html]: @bmnepali please provide all "greps" and folders these greps been run through as discussed.

Comment (bmnepali) [reply uk.co.compucorp.civicrm.tasksassignments/views/contact/documents.html]: @igorpavlov above grep commands were ran from within `views/` folders that stores templates for the app.

**Comment by bmnepali** [comment]
Fixed merge conflicts by git rebase.
